### DISCUSSION TITLE
/assets/register should return url for asset

### DISCUSSION THREAD

**Discussion by bartektrybala** [amzn/ads-advanced-tools-docs#92] [Ideas]
It will be a lot of easier for api users when Amazon Ads would return proper url for asset after registration. We want to use it in a preview feature, and now we have to do another GET request to fetch proper urls.

btw: It’s possible to extract url for asset preview from the upload url, but for production env it’s not recommended.
So, is there any problem to return that in register response

**Comment by 18975035535** [comment]
@bartektrybala 我在用注册资产的接口时候一直提示：AARServiceCorruptAssetException while executing registration workflow: Exception while executing registratio，请问你在使用的时候有没有碰到这种问题

**Reply by bartektrybala** [reply]
?

**Comment by elizafarley** [comment]
@bartektrybala Thanks for this suggestion! To clarify, are you looking for a way to preview an asset using the API?

**Reply by bartektrybala** [reply]
Yeah
